tests/unit/s2n_server_key_share_extension_test.c

```diff
@@ -555,21 +555,32 @@ int main(int argc, char **argv)
 
             /* If PQ is disabled, the client will not have sent PQ IDs/keyshares in the ClientHello;
              * if the server responded with a PQ keyshare, we should error. */
-            if (!s2n_pq_is_enabled()) {
-                struct s2n_connection *client_conn = NULL;
-                EXPECT_NOT_NULL(client_conn = s2n_connection_new(S2N_CLIENT));
+            {
+                DEFER_CLEANUP(struct s2n_connection *client_conn = s2n_connection_new(S2N_CLIENT),
+                        s2n_connection_ptr_free);
+                EXPECT_NOT_NULL(client_conn);
                 client_conn->security_policy_override = &test_security_policy;
 
+                /* The connection is set up to be receiving a HelloRetryRequest in order to exit
+                 * s2n_server_key_share_extension.recv earlier for testing the success case.
+                 */
+                client_conn->handshake.state_machine = S2N_STATE_MACHINE_TLS13;
+                EXPECT_SUCCESS(s2n_set_connection_hello_retry_flags(client_conn));
+                EXPECT_TRUE(s2n_is_hello_retry_message(client_conn));
```

Comment:
Is this necessary? What happens if you don't specifically test a retry?

Reply:
I'm just trying to test that the cause of the error was due to PQ being disabled, so I'm abusing the retry check to exit the function successfully earlier: 
https://github.com/aws/s2n-tls/blob/ada65a76ac7154644f85024ed5ae4696f4409226/tls/extensions/s2n_server_key_share.c#L195-L197

Without this, the connection would have to be in a state that can actually process the key share extension, which seemed more complicated:
https://github.com/aws/s2n-tls/blob/ada65a76ac7154644f85024ed5ae4696f4409226/tls/extensions/s2n_server_key_share.c#L199-L203

Maybe this isn't a good idea though? I can look into other ways to test this if it's not good.

Reply:
Tricky. It seems a little brittle, but probably fine? It's definitely not worse than it was before.